server/metrics/collectors/collectors.go

```diff
@@ -17,46 +17,87 @@ const (
+	// Deprecated: in favour of ImagePullsSkippedBytesTotal
 	ImagePullsByNameSkipped Collector = crioPrefix + "image_pulls_by_name_skipped"
 
 	// ImagePullsFailures is the key for failed image downloads in CRI-O.
+	// Deprecated: in favour of ImagePullsFailureTotal
 	ImagePullsFailures Collector = crioPrefix + "image_pulls_failures"
 
 	// ImagePullsSuccesses is the key for successful image downloads in CRI-O.
+	// Deprecated: in favour of ImagePullsSuccessTotal
 	ImagePullsSuccesses Collector = crioPrefix + "image_pulls_successes"
 
 	// ImagePullsLayerSize is the key for CRI-O image pull metrics per layer.
 	ImagePullsLayerSize Collector = crioPrefix + "image_pulls_layer_size"
 
 	// ImageLayerReuse is the key for the CRI-O image layer reuse metrics.
+	// Deprecated: in favour of ImageLayerReuseTotal
 	ImageLayerReuse Collector = crioPrefix + "image_layer_reuse"
 
 	// ContainersOOMTotal is the key for the total CRI-O container out of memory metrics.
 	ContainersOOMTotal Collector = crioPrefix + "containers_oom_total"
 
 	// ContainersOOM is the key for the CRI-O container out of memory metrics per container name.
+	// Deprecated: in favour of ContainersOOMCountTotal
 	ContainersOOM Collector = crioPrefix + "containers_oom"
 
 	// ProcessesDefunct is the key for the total number of defunct processes in a node.
 	ProcessesDefunct Collector = crioPrefix + "processes_defunct"
+
+	// OperationsTotal is the key for CRI-O operation metrics.
+	OperationsTotal Collector = crioPrefix + "operations_total"
+
+	// OperationsLatencySeconds is the key for the operation latency metrics for each CRI call.
+	OperationsLatencySeconds Collector = crioPrefix + "operations_latency_seconds"
+
+	// OperationsLatencySecondsTotal is the key for the operation latency metrics.
+	OperationsLatencySecondsTotal Collector = crioPrefix + "operations_latency_seconds_total"
+
+	// OperationsErrorsTotal is the key for the operation error metrics.
+	OperationsErrorsTotal Collector = crioPrefix + "operations_errors_total"
+
+	// ImagePullsBytesTotal is the key for CRI-O image pull metrics.
+	ImagePullsBytesTotal Collector = crioPrefix + "image_pulls_bytes_total"
+
+	// ImagePullsSkippedBytesTotal is the key for CRI-O skipped image pull metrics.
+	ImagePullsSkippedBytesTotal Collector = crioPrefix + "image_pulls_skipped_bytes_total"
+
+	// ImagePullsFailureTotal is the key for failed image downloads in CRI-O.
+	ImagePullsFailureTotal Collector = crioPrefix + "image_pulls_failure_total"
+
+	// ImagePullsSuccessTotal is the key for successful image downloads in CRI-O.
+	ImagePullsSuccessTotal Collector = crioPrefix + "image_pulls_success_total"
+
+	// ImageLayerReuseTotal is the key for the CRI-O image layer reuse metrics.
+	ImageLayerReuseTotal Collector = crioPrefix + "image_layer_reuse_total"
+
+	// ContainersOOMCountTotal is the key for the CRI-O container out of memory metrics per container name.
+	ContainersOOMCountTotal Collector = crioPrefix + "containers_oom_count_total"
```

Comment:
1: Should we mention in docs or comments the possibility of oom metric by name having high cardinality, but the expectation that it shouldn't and that the metric is worth keeping?
2: shouldn't we mention `by name` or something. `containers` implies it's a total count for all containers. maybe I'm misunderstanding the purpose of this one though

Reply:
Yeah: we can mention in the docs about the high cardinality but why we intend to keep it with the reasonable assumption that number of containers OOMing would be low ideally.

Reply:
We're not keeping `_by_name` in the metric name as previously discussed in above thread(s) that `_by_name` part inclusion in metric is bad naming.

Reply:
Added 1. in https://github.com/cri-o/cri-o/blob/8ec8b7396e018ad08f28b159a235aaff06d6287e/server/metrics/metrics.go#L308-L310